# COGS 108 - Data Checkpoint

## Authors

Fawaz Alsenayin: Data curation
Ghasaq Al Khalaf: Background research, Writing - original draft 
Hadia Beshir: Conceptualization
Shelby Giang: Writing - review & editing
Nawar Kidwai: Background research

## Research Question

Does higher dietary linoleic acid intake, commonly found in fast-food and ultra-processed diets using seed oils, increase obesity prevalence and cardiovascular risk, measured by BMI and total cholesterol, among U.S. adults?

## Background and Prior Work

Dietary patterns in the United States have shifted toward greater consumption of ultra-processed foods, many of which are energy-dense and contain refined vegetable oils. Linoleic acid is an omega-6 polyunsaturated fatty acid commonly found in oils such as corn, sunflower, soybean, and canola oil.[^1] According to the CDC’s National Center for Health Statistics, ultra-processed foods made up about 55% of total daily calorie intake among people age 1 and older in the United States from August 2021 to August 2023, with an even higher percentage among youth.[^1] Prior research has also found that higher consumption of ultra-processed foods is associated with increased risk of obesity, type 2 diabetes, and cardiovascular disease.[^2] However, research on linoleic acid itself is more mixed, with some evidence suggesting that replacing saturated fats with linoleic-acid-rich unsaturated fats may improve cardiovascular outcomes.[^3] Because of this, our project focuses on whether higher dietary linoleic acid intake is associated with obesity and cardiovascular risk markers, rather than assuming it is directly harmful.

This topic is important because many Americans may not know how much linoleic acid they consume through fast-food and ultra-processed diets. Over the 20th century, the U.S. diet shifted away from traditional animal fats such as butter and lard toward greater use of industrial vegetable oils, which increased dietary exposure to omega-6 fatty acids.[^4] Omega-6 fatty acids are essential in the diet, but some researchers argue that a high omega-6 to omega-3 ratio may contribute to inflammation-related health concerns.[^4] Omega-3 fatty acids, which are found in foods such as salmon, sardines, tuna, chia seeds, and flaxseeds, may help balance inflammatory processes in the body.[^4] This background helps explain why linoleic acid intake is debated and why it is useful to study its relationship with measurable health outcomes.

For our project, we will focus on obesity and cardiovascular risk because these outcomes can be measured using NHANES variables. BMI will be used as a marker of obesity risk, with BMI of 30 or higher commonly used to classify obesity. Total cholesterol will be used as a cardiovascular risk marker because high cholesterol is associated with greater risk of heart disease. However, these measures have limitations: BMI does not fully capture body composition, and total cholesterol alone does not fully diagnose cardiovascular disease. Therefore, our analysis will examine whether higher linoleic acid intake is associated with BMI and total cholesterol, while clearly stating that the results do not prove causation.

References:
[^1]: Williams, A. M., Couch, C. A., Emmerich, S. E., & Ogburn, D. F. (2025). *Ultra-processed food consumption among youth and adults: United States, August 2021–August 2023*. NCHS Data Brief, No. 536. Centers for Disease Control and Prevention. https://www.cdc.gov/nchs/products/databriefs/db536.htm

[^2]: Pagliai, G., Dinu, M., Madarena, M. P., Bonaccio, M., Iacoviello, L., & Sofi, F. (2021). *Consumption of ultra-processed foods and health status: A systematic review and meta-analysis*. *British Journal of Nutrition, 125*(3), 308–318. https://www.cambridge.org/core/journals/british-journal-of-nutrition/article/consumption-of-ultraprocessed-foods-and-health-status-a-systematic-review-and-metaanalysis/

[^3]: Johns Hopkins Bloomberg School of Public Health. (2025). *The evidence behind seed oils’ health effects*. https://publichealth.jhu.edu/2025/the-evidence-behind-seed-oils-health-effects

[^4]: Simopoulos, A. P. (2002). The importance of the ratio of omega-6/omega-3 essential fatty acids. *Biomedicine & Pharmacotherapy, 56*(8), 365–379. https://doi.org/10.1016/S0753-3322(02)00253-6

## Hypothesis


We hypothesize that higher dietary linoleic acid intake, commonly found in fast-food and ultra-processed diets that use seed oils, will be associated with higher obesity and cardiovascular risk among U.S. adults. More specifically, we predict that adults with higher linoleic acid intake will have higher average BMI, higher obesity prevalence, and higher total cholesterol levels compared to adults with lower linoleic acid intake. This prediction is based on prior research showing that ultra-processed food consumption is associated with obesity and cardiovascular disease risk, while our project uses linoleic acid intake as a measurable proxy for seed-oil-rich dietary patterns.

## Data

### Data Overview

We are using four NHANES August 2021–August 2023 datasets from the CDC/National Center for Health Statistics. These datasets are publicly available as XPT files and do not require special permission. We will combine them using `SEQN`, the respondent sequence number, so that each row in our final dataset represents one adult participant.

**Dataset #1**
- **Dataset Name:** Dietary Interview - Total Nutrient Intakes, First Day (`DR1TOT_L`)
- **Link to the dataset:** https://wwwn.cdc.gov/Nchs/Data/Nhanes/Public/2021/DataFiles/DR1TOT_L.htm
- **Number of observations:** 8,860
- **Number of variables:** 168
- **Relevant variables:** `SEQN` for participant ID, `DR1TP182` for linoleic acid intake in grams, `DR1TKCAL` for total calories in kcal, and `DR1TTFAT` for total fat in grams.
- **Shortcomings:** This dataset uses a 24-hour dietary recall, so one day may not represent a participant’s usual diet. It also does not directly identify seed oil or fast-food intake, so we use linoleic acid intake as a proxy.

**Dataset #2**
- **Dataset Name:** Body Measures (`BMX_L`)
- **Link to the dataset:** https://wwwn.cdc.gov/Nchs/Data/Nhanes/Public/2021/DataFiles/BMX_L.htm
- **Number of observations:** 8,860
- **Number of variables:** 22
- **Relevant variables:** `SEQN` for participant ID and `BMXBMI` for Body Mass Index in kg/m².
- **Shortcomings:** BMI is useful for measuring obesity risk, but it does not fully describe body composition or overall health.

**Dataset #3**
- **Dataset Name:** Cholesterol - Total (`TCHOL_L`)
- **Link to the dataset:** https://wwwn.cdc.gov/Nchs/Data/Nhanes/Public/2021/DataFiles/TCHOL_L.htm
- **Number of observations:** 8,068
- **Number of variables:** 4
- **Relevant variables:** `SEQN` for participant ID and `LBXTC` for total cholesterol in mg/dL.
- **Shortcomings:** Total cholesterol is only one cardiovascular risk marker. It does not fully measure cardiovascular disease risk by itself.

**Dataset #4**
- **Dataset Name:** Demographic Variables and Sample Weights (`DEMO_L`)
- **Link to the dataset:** https://wwwn.cdc.gov/Nchs/Data/Nhanes/Public/2021/DataFiles/DEMO_L.htm
- **Number of observations:** 11,933
- **Number of variables:** 27
- **Relevant variables:** `SEQN` for participant ID, `RIDAGEYR` for age, `RIAGENDR` for sex, `RIDRETH3` for race/ethnicity, and `INDFMPIR` for income-to-poverty ratio.
- **Shortcomings:** Some demographic variables are limited for privacy reasons, and this dataset does not include every possible lifestyle factor that may affect BMI or cholesterol.

We plan to merge these datasets using `SEQN`. After merging, we will filter to adults, remove missing values in the main analysis variables, and create new variables for obesity status and high cholesterol status.

In [1]:
# Run this code every time when you're actively developing modules in .py files.  It's not needed if you aren't making modules
#
## this code is necessary for making sure that any modules we load are updated here 
## when their source code .py files are modified

%load_ext autoreload
%autoreload 2

In [2]:
import pandas as pd
import numpy as np
from pathlib import Path

# Setup code -- this only needs to be run once after cloning the repo!
# this code downloads the data from its source to the `data/00-raw/` directory
# if the data hasn't updated you don't need to do this again!

# if you don't already have these packages (you should!) uncomment this line
# %pip install requests tqdm

import sys
sys.path.append('./modules') # this tells python where to look for modules to import

import get_data # this is where we get the function we need to download data

# replace the urls and filenames in this list with your actual datafiles
# yes you can use Google drive share links or whatever
# format is a list of dictionaries; 
# each dict has keys of 
#   'url' where the resource is located
#   'filename' for the local filename where it will be stored 
datafiles = [
    { "url": "https://wwwn.cdc.gov/Nchs/Data/Nhanes/Public/2021/DataFiles/DR1TOT_L.xpt", "filename": "DR1TOT_L.xpt"},
    { "url": "https://wwwn.cdc.gov/Nchs/Data/Nhanes/Public/2021/DataFiles/BMX_L.xpt", "filename": "BMX_L.xpt"},
    { "url": "https://wwwn.cdc.gov/Nchs/Data/Nhanes/Public/2021/DataFiles/TCHOL_L.xpt", "filename": "TCHOL_L.xpt"},
    { "url": "https://wwwn.cdc.gov/Nchs/Data/Nhanes/Public/2021/DataFiles/DEMO_L.xpt", "filename": "DEMO_L.xpt"}
]

get_data.get_raw(datafiles,destination_directory='data/00-raw/')

raw_path = Path("data/00-raw")
processed_path = Path("data/02-processed")
processed_path.mkdir(parents=True, exist_ok=True)

diet = pd.read_sas(raw_path / "DR1TOT_L.xpt")
bmx = pd.read_sas(raw_path / "BMX_L.xpt")
chol = pd.read_sas(raw_path / "TCHOL_L.xpt")
demo = pd.read_sas(raw_path / "DEMO_L.xpt")

Overall Download Progress:  25%|██▌       | 1/4 [00:01<00:05,  1.69s/it]

Successfully downloaded: DR1TOT_L.xpt



Overall Download Progress:  50%|█████     | 2/4 [00:01<00:01,  1.26it/s]

Successfully downloaded: BMX_L.xpt



Overall Download Progress:  75%|███████▌  | 3/4 [00:02<00:00,  2.00it/s]

Successfully downloaded: TCHOL_L.xpt



Overall Download Progress: 100%|██████████| 4/4 [00:02<00:00,  1.74it/s]


Successfully downloaded: DEMO_L.xpt
DR1TOT_L shape: (8860, 168)
BMX_L shape: (8860, 22)
TCHOL_L shape: (8068, 4)
DEMO_L shape: (11933, 27)


### Dataset 1 - Day 1 Dietary Nutrient Intake

The `DR1TOT_L` dataset contains Day 1 dietary recall data from NHANES August 2021–August 2023. Each row represents one participant who completed a 24-hour dietary recall interview, and each column represents a nutrient or dietary measure. This dataset has 8,860 observations and 168 variables.

The most important variable for our project is `DR1TP182`, which measures PFA 18:2 octadecadienoic acid intake in grams. We will use this as our measure of linoleic acid intake because linoleic acid is commonly found in seed oils used in fast-food and ultra-processed diets. We will also use `DR1TKCAL`, which measures total calorie intake in kilocalories, and `DR1TTFAT`, which measures total fat intake in grams. These variables help us understand whether higher linoleic acid intake is related to overall diet size and fat consumption.

A major limitation is that this dataset does not directly measure “seed oil intake” or identify the exact oils used in the foods participants ate. The dietary information is also based on only one 24-hour recall, so it may not fully represent a participant’s long-term eating pattern. Because of this, we will interpret linoleic acid intake as a proxy for seed-oil-rich diets and focus on associations rather than causation.

In [3]:
# Dataset 1 - Day 1 Dietary Nutrient Intake

# Keep only variables relevant to our project
diet_clean = diet[["SEQN", "DR1TP182", "DR1TKCAL", "DR1TTFAT"]].copy()

# Rename columns for readability
diet_clean = diet_clean.rename(columns={
    "DR1TP182": "linoleic_acid_g",
    "DR1TKCAL": "calories_kcal",
    "DR1TTFAT": "total_fat_g"
})

# Demonstrate size of dataset
print("Original diet dataset shape:", diet.shape)
print("Selected diet dataset shape:", diet_clean.shape)

# Check that each row is one participant
print("Duplicate participant IDs:", diet_clean["SEQN"].duplicated().sum())

# Check missing values
missing_table = pd.DataFrame({
    "missing_count": diet_clean.isna().sum(),
    "missing_percent": diet_clean.isna().mean() * 100
})

display(missing_table)

# Summary statistics to check ranges and possible outliers
display(diet_clean.describe())

# Flag suspicious values
suspicious_diet = diet_clean[
    (diet_clean["linoleic_acid_g"] < 0) |
    (diet_clean["calories_kcal"] <= 0) |
    (diet_clean["total_fat_g"] < 0)
]

print("Number of suspicious entries:", suspicious_diet.shape[0])
display(suspicious_diet.head())

# Clean the data by dropping rows missing required dietary variables
diet_clean = diet_clean.dropna(subset=[
    "linoleic_acid_g",
    "calories_kcal",
    "total_fat_g"
])

# Remove impossible values
diet_clean = diet_clean[
    (diet_clean["linoleic_acid_g"] >= 0) &
    (diet_clean["calories_kcal"] > 0) &
    (diet_clean["total_fat_g"] >= 0)
].copy()

# Save cleaned Dataset #1
diet_clean.to_csv(processed_path / "diet_clean.csv", index=False)

print("Final cleaned diet dataset shape:", diet_clean.shape)
display(diet_clean.head())

Original diet dataset shape: (8860, 168)
Selected diet dataset shape: (8860, 4)
Duplicate participant IDs: 0


,missing_count,missing_percent
SEQN,0,0.000000
linoleic_acid_g,2166,24.446953
calories_kcal,2166,24.446953
total_fat_g,2166,24.446953


,SEQN,linoleic_acid_g,calories_kcal,total_fat_g
count,8860.000000,6.694000e+03,6.694000e+03,6.694000e+03
mean,136345.828217,1.625569e+01,1.922305e+03,7.995149e+01
std,3453.776310,1.124088e+01,8.904012e+02,4.464114e+01
min,130378.000000,5.397605e-79,5.397605e-79,5.397605e-79
25%,133319.750000,8.719500e+00,1.332000e+03,4.956250e+01
50%,136377.500000,1.393550e+01,1.787000e+03,7.282000e+01
75%,139336.250000,2.097975e+01,2.367000e+03,1.013350e+02
max,142310.000000,1.714240e+02,1.044600e+04,5.056200e+02


Number of suspicious entries: 0


,SEQN,linoleic_acid_g,calories_kcal,total_fat_g


Final cleaned diet dataset shape: (6694, 4)


,SEQN,linoleic_acid_g,calories_kcal,total_fat_g
0,130378.0,9.905,1740.0,55.07
1,130379.0,22.948,2741.0,67.18
2,130380.0,10.397,1995.0,65.38
3,130381.0,14.641,1676.0,63.57
4,130382.0,12.334,1222.0,39.80


The original dietary dataset had 8,860 rows and 168 variables. After selecting the variables needed for our project, the dataset had 8,860 rows and 4 variables: participant ID, linoleic acid intake, total calories, and total fat. There were no duplicate participant IDs, so the dataset is tidy because each row represents one participant. About 24.45% of rows were missing the main dietary variables, likely because those participants did not have complete Day 1 dietary recall data. We dropped those rows because linoleic acid, calories, and total fat are required for our analysis. After cleaning, the dietary dataset had 6,694 observations and 4 variables.

### Dataset 2 - Body Measures

The `BMX_L` dataset contains body measurement data from NHANES August 2021–August 2023. Each row represents one participant, and each column represents a body measurement collected during the NHANES physical examination. This dataset has 8,860 observations and 22 variables.

The most important variable for our project is `BMXBMI`, which measures Body Mass Index, or BMI, in kg/m². BMI is commonly used to classify adult weight status. A BMI of 30 or higher is commonly used to classify obesity, so this variable will help us measure obesity prevalence in our project.

A limitation of BMI is that it does not fully measure body composition. For example, it does not distinguish between fat mass and muscle mass. However, BMI is still useful for our project because it is a common population-level marker of obesity risk.

In [4]:
# Dataset #2: Body Measures

# Keep only variables relevant to our project
bmx_clean = bmx[["SEQN", "BMXBMI"]].copy()

# Rename columns for readability
bmx_clean = bmx_clean.rename(columns={
    "BMXBMI": "bmi"
})

# Demonstrate size of dataset
print("Original body measures dataset shape:", bmx.shape)
print("Selected body measures dataset shape:", bmx_clean.shape)

# Check that each row is one participant
print("Duplicate participant IDs:", bmx_clean["SEQN"].duplicated().sum())

# Check missing values
missing_table_bmx = pd.DataFrame({
    "missing_count": bmx_clean.isna().sum(),
    "missing_percent": bmx_clean.isna().mean() * 100
})

display(missing_table_bmx)

# Summary statistics to check ranges and possible outliers
display(bmx_clean.describe())

# Flag suspicious BMI values
suspicious_bmx = bmx_clean[
    (bmx_clean["bmi"] <= 0) |
    (bmx_clean["bmi"] > 100)
]

print("Number of suspicious entries:", suspicious_bmx.shape[0])
display(suspicious_bmx.head())

# Clean the data by dropping rows missing BMI
bmx_clean = bmx_clean.dropna(subset=["bmi"])

# Remove impossible or extreme suspicious values
bmx_clean = bmx_clean[
    (bmx_clean["bmi"] > 0) &
    (bmx_clean["bmi"] <= 100)
].copy()

# Create obesity variable
bmx_clean["obese"] = bmx_clean["bmi"] >= 30

# Save cleaned Dataset #2
bmx_clean.to_csv(processed_path / "bmx_clean.csv", index=False)

print("Final cleaned body measures dataset shape:", bmx_clean.shape)
display(bmx_clean.head())

Original body measures dataset shape: (8860, 22)
Selected body measures dataset shape: (8860, 2)
Duplicate participant IDs: 0


,missing_count,missing_percent
SEQN,0,0.000000
bmi,389,4.390519


,SEQN,bmi
count,8860.000000,8471.000000
mean,136345.828217,27.246665
std,3453.776310,8.137781
min,130378.000000,11.100000
25%,133319.750000,21.600000
50%,136377.500000,26.400000
75%,139336.250000,31.700000
max,142310.000000,74.800000


Number of suspicious entries: 0


,SEQN,bmi


Final cleaned body measures dataset shape: (8471, 3)


,SEQN,bmi,obese
0,130378.0,27.0,False
1,130379.0,33.5,True
2,130380.0,29.7,False
3,130381.0,23.8,False
5,130386.0,30.2,True


The original body measures dataset had 8,860 rows and 22 variables. After selecting the variables needed for our project, the dataset had 8,860 rows and 2 variables: participant ID and BMI. There were no duplicate participant IDs, so the dataset is tidy because each row represents one participant. BMI was missing for 389 participants, which is about 4.39% of the dataset, so we dropped those rows because BMI is required to measure obesity risk. There were no suspicious BMI values based on our cutoff, and the final cleaned body measures dataset had 8,471 observations and 3 variables after adding the obesity indicator.

### Dataset 3 - Total Cholesterol

The `TCHOL_L` dataset contains total cholesterol lab data from NHANES August 2021–August 2023. Each row represents one participant, and each column represents a cholesterol-related lab variable. This dataset has 8,068 observations and 4 variables.

The most important variable for our project is `LBXTC`, which measures total cholesterol in mg/dL. Total cholesterol is a common cardiovascular risk marker. In general, total cholesterol below 200 mg/dL is considered desirable, while 200 mg/dL or higher may indicate higher cardiovascular risk. We will use this variable to measure cardiovascular risk in relation to linoleic acid intake.

A limitation of this dataset is that total cholesterol alone does not fully measure cardiovascular disease risk. Other factors such as HDL cholesterol, LDL cholesterol, triglycerides, blood pressure, smoking, and diabetes status also affect cardiovascular risk. Because of this, we will describe total cholesterol as a cardiovascular risk marker, not as a complete diagnosis of cardiovascular disease.

In [5]:
# Dataset 3: Total Cholesterol

# Keep only variables relevant to our project
chol_clean = chol[["SEQN", "LBXTC"]].copy()

# Rename columns for readability
chol_clean = chol_clean.rename(columns={
    "LBXTC": "total_cholesterol_mg_dl"
})

# Demonstrate size of dataset
print("Original cholesterol dataset shape:", chol.shape)
print("Selected cholesterol dataset shape:", chol_clean.shape)

# Check that each row is one participant
print("Duplicate participant IDs:", chol_clean["SEQN"].duplicated().sum())

# Check missing values
missing_table_chol = pd.DataFrame({
    "missing_count": chol_clean.isna().sum(),
    "missing_percent": chol_clean.isna().mean() * 100
})

display(missing_table_chol)

# Summary statistics to check ranges and possible outliers
display(chol_clean.describe())

# Flag suspicious cholesterol values
suspicious_chol = chol_clean[
    (chol_clean["total_cholesterol_mg_dl"] <= 0) |
    (chol_clean["total_cholesterol_mg_dl"] > 500)
]

print("Number of suspicious entries:", suspicious_chol.shape[0])
display(suspicious_chol.head())

# Clean the data by dropping rows missing total cholesterol
chol_clean = chol_clean.dropna(subset=["total_cholesterol_mg_dl"])

# Remove impossible or extreme suspicious values
chol_clean = chol_clean[
    (chol_clean["total_cholesterol_mg_dl"] > 0) &
    (chol_clean["total_cholesterol_mg_dl"] <= 500)
].copy()

# Create high cholesterol variable
chol_clean["high_cholesterol"] = chol_clean["total_cholesterol_mg_dl"] >= 200

# Save cleaned Dataset #3
chol_clean.to_csv(processed_path / "chol_clean.csv", index=False)

print("Final cleaned cholesterol dataset shape:", chol_clean.shape)
display(chol_clean.head())

Original cholesterol dataset shape: (8068, 4)
Selected cholesterol dataset shape: (8068, 2)
Duplicate participant IDs: 0


,missing_count,missing_percent
SEQN,0,0.000000
total_cholesterol_mg_dl,1178,14.600892


,SEQN,total_cholesterol_mg_dl
count,8068.000000,6890.000000
mean,136360.136589,181.541074
std,3448.491865,42.316140
min,130378.000000,62.000000
25%,133345.500000,151.000000
50%,136391.000000,178.000000
75%,139336.250000,207.000000
max,142310.000000,438.000000


Number of suspicious entries: 0


,SEQN,total_cholesterol_mg_dl


Final cleaned cholesterol dataset shape: (6890, 3)


,SEQN,total_cholesterol_mg_dl,high_cholesterol
0,130378.0,264.0,True
1,130379.0,214.0,True
2,130380.0,187.0,False
3,130386.0,183.0,False
4,130387.0,203.0,True


The original cholesterol dataset had 8,068 rows and 4 variables. After selecting the variables needed for our project, the dataset had 8,068 rows and 2 variables: participant ID and total cholesterol. There were no duplicate participant IDs, so the dataset is tidy because each row represents one participant. Total cholesterol was missing for 1,178 participants, which is about 14.60% of the dataset, so we dropped those rows because cholesterol is required for our cardiovascular risk analysis. There were no suspicious cholesterol values based on our cutoff, and the final cleaned cholesterol dataset had 6,890 observations and 3 variables after adding the high cholesterol indicator.

### Dataset 4 - Demographics

The `DEMO_L` dataset contains demographic information from NHANES August 2021–August 2023. Each row represents one participant, and each column represents a demographic or survey-design variable. This dataset has 11,933 observations and 27 variables.

The most important variables for our project are `RIDAGEYR`, which measures age in years; `RIAGENDR`, which represents sex; `RIDRETH3`, which represents race/ethnicity; and `INDFMPIR`, which represents family income-to-poverty ratio. These variables are important because age, sex, race/ethnicity, and income may affect both diet and health outcomes. We will also use age to filter the dataset to adults.

A limitation of this dataset is that it does not include every possible factor that may influence BMI or cholesterol. For example, physical activity, smoking, medication use, and medical history may also affect obesity and cardiovascular risk. Some demographic variables may also be missing or grouped broadly to protect participant privacy.

In [6]:
# Dataset 4: Demographics

# Keep only variables relevant to our project
demo_clean = demo[["SEQN", "RIDAGEYR", "RIAGENDR", "RIDRETH3", "INDFMPIR"]].copy()

# Rename columns for readability
demo_clean = demo_clean.rename(columns={
    "RIDAGEYR": "age",
    "RIAGENDR": "sex",
    "RIDRETH3": "race_ethnicity",
    "INDFMPIR": "income_poverty_ratio"
})

# Demonstrate size of dataset
print("Original demographics dataset shape:", demo.shape)
print("Selected demographics dataset shape:", demo_clean.shape)

# Check that each row is one participant
print("Duplicate participant IDs:", demo_clean["SEQN"].duplicated().sum())

# Check missing values
missing_table_demo = pd.DataFrame({
    "missing_count": demo_clean.isna().sum(),
    "missing_percent": demo_clean.isna().mean() * 100
})

display(missing_table_demo)

# Summary statistics to check ranges and possible outliers
display(demo_clean.describe())

# Flag suspicious demographic values
suspicious_demo = demo_clean[
    (demo_clean["age"] < 0) |
    (~demo_clean["sex"].isin([1, 2])) |
    (~demo_clean["race_ethnicity"].isin([1, 2, 3, 4, 6, 7])) |
    (demo_clean["income_poverty_ratio"] < 0)
]

print("Number of suspicious entries:", suspicious_demo.shape[0])
display(suspicious_demo.head())

# Clean the data:
# Keep adults only because our research question focuses on U.S. adults.
demo_clean = demo_clean[demo_clean["age"] >= 20].copy()

# Drop rows missing required demographic variables
demo_clean = demo_clean.dropna(subset=[
    "age",
    "sex",
    "race_ethnicity"
])

# Income-to-poverty ratio is useful but may be missing, so we keep missing income values for now.
# We may drop or handle income missingness later depending on the final model.

# Remove impossible demographic values
demo_clean = demo_clean[
    (demo_clean["age"] >= 20) &
    (demo_clean["sex"].isin([1, 2])) &
    (demo_clean["race_ethnicity"].isin([1, 2, 3, 4, 6, 7]))
].copy()

# Save cleaned Dataset #4
demo_clean.to_csv(processed_path / "demo_clean.csv", index=False)

print("Final cleaned demographics dataset shape:", demo_clean.shape)
display(demo_clean.head())

Original demographics dataset shape: (11933, 27)
Selected demographics dataset shape: (11933, 5)
Duplicate participant IDs: 0


,missing_count,missing_percent
SEQN,0,0.00000
age,0,0.00000
sex,0,0.00000
race_ethnicity,0,0.00000
income_poverty_ratio,2041,17.10383


,SEQN,age,sex,race_ethnicity,income_poverty_ratio
count,11933.000000,1.193300e+04,11933.000000,11933.000000,9.892000e+03
mean,136344.000000,3.831786e+01,1.532808,3.320540,2.708174e+00
std,3444.904716,2.560199e+01,0.498943,1.518379,1.670119e+00
min,130378.000000,5.397605e-79,1.000000,1.000000,5.397605e-79
25%,133361.000000,1.300000e+01,1.000000,3.000000,1.180000e+00
50%,136344.000000,3.700000e+01,2.000000,3.000000,2.500000e+00
75%,139327.000000,6.200000e+01,2.000000,4.000000,4.500000e+00
max,142310.000000,8.000000e+01,2.000000,7.000000,5.000000e+00


Number of suspicious entries: 0


,SEQN,age,sex,race_ethnicity,income_poverty_ratio


Final cleaned demographics dataset shape: (7809, 5)


,SEQN,age,sex,race_ethnicity,income_poverty_ratio
0,130378.0,43.0,1.0,6.0,5.00
1,130379.0,66.0,1.0,3.0,5.00
2,130380.0,44.0,2.0,2.0,1.41
6,130384.0,43.0,1.0,1.0,0.63
7,130385.0,65.0,2.0,3.0,5.00


The original demographics dataset had 11,933 rows and 27 variables. After selecting the variables needed for our project, the dataset had 11,933 rows and 5 variables: participant ID, age, sex, race/ethnicity, and income-to-poverty ratio. There were no duplicate participant IDs, so the dataset is tidy because each row represents one participant. Age, sex, and race/ethnicity had no missing values, but income-to-poverty ratio was missing for 2,041 participants, or about 17.10% of the dataset. We kept missing income values for now because income is useful but not required for the main analysis, then filtered the dataset to adults age 20 and older. The final cleaned demographics dataset had 7,809 observations and 5 variables.

## Ethics

## Ethics 

Instructions: Keep the contents of this cell. For each item on the checklist
-  put an X there if you've considered the item
-  IF THE ITEM IS RELEVANT place a short paragraph after the checklist item discussing the issue.
  
Items on this checklist are meant to provoke discussion among good-faith actors who take their ethical responsibilities seriously. Your teams will document these discussions and decisions for posterity using this section.  You don't have to solve these problems, you just have to acknowledge any potential harm no matter how unlikely.

Here is a [list of real world examples](https://deon.drivendata.org/examples/) for each item in the checklist that can refer to.

[![Deon badge](https://img.shields.io/badge/ethics%20checklist-deon-brightgreen.svg?style=popout-square)](http://deon.drivendata.org/)

### A. Data Collection
 - [X] **A.1 Informed consent**: If there are human subjects, have they given informed consent, where subjects affirmatively opt-in and have a clear understanding of the data uses to which they consent?

> Example of how to use the checkbox, and also of how you can put in a short paragraph that discusses the way this checklist item affects your project.  Remove this paragraph and the X in the checkbox before you fill this out for your project

 - [X] **A.2 Collection bias**: Have we considered sources of bias that could be introduced during data collection and survey design and taken steps to mitigate those?
 - [X] **A.3 Limit PII exposure**: Have we considered ways to minimize exposure of personally identifiable information (PII) for example through anonymization or not collecting information that isn't relevant for analysis?
 - [X] **A.4 Downstream bias mitigation**: Have we considered ways to enable testing downstream results for biased outcomes (e.g., collecting data on protected group status like race or gender)?

### B. Data Storage
 - [X] **B.1 Data security**: Do we have a plan to protect and secure data (e.g., encryption at rest and in transit, access controls on internal users and third parties, access logs, and up-to-date software)?
 - [ ] **B.2 Right to be forgotten**: Do we have a mechanism through which an individual can request their personal information be removed?
 - [X] **B.3 Data retention plan**: Is there a schedule or plan to delete the data after it is no longer needed?

### C. Analysis
 - [X] **C.1 Missing perspectives**: Have we sought to address blindspots in the analysis through engagement with relevant stakeholders (e.g., checking assumptions and discussing implications with affected communities and subject matter experts)?
 - [X] **C.2 Dataset bias**: Have we examined the data for possible sources of bias and taken steps to mitigate or address these biases (e.g., stereotype perpetuation, confirmation bias, imbalanced classes, or omitted confounding variables)?
 - [X] **C.3 Honest representation**: Are our visualizations, summary statistics, and reports designed to honestly represent the underlying data?
 - [X] **C.4 Privacy in analysis**: Have we ensured that data with PII are not used or displayed unless necessary for the analysis?
 - [X] **C.5 Auditability**: Is the process of generating the analysis well documented and reproducible if we discover issues in the future?

### D. Modeling
 - [X] **D.1 Proxy discrimination**: Have we ensured that the model does not rely on variables or proxies for variables that are unfairly discriminatory?
 - [X] **D.2 Fairness across groups**: Have we tested model results for fairness with respect to different affected groups (e.g., tested for disparate error rates)?
 - [X] **D.3 Metric selection**: Have we considered the effects of optimizing for our defined metrics and considered additional metrics?
 - [X] **D.4 Explainability**: Can we explain in understandable terms a decision the model made in cases where a justification is needed?
 - [X] **D.5 Communicate limitations**: Have we communicated the shortcomings, limitations, and biases of the model to relevant stakeholders in ways that can be generally understood?

### E. Deployment
 - [ ] **E.1 Monitoring and evaluation**: Do we have a clear plan to monitor the model and its impacts after it is deployed (e.g., performance monitoring, regular audit of sample predictions, human review of high-stakes decisions, reviewing downstream impacts of errors or low-confidence decisions, testing for concept drift)?
 - [ ] **E.2 Redress**: Have we discussed with our organization a plan for response if users are harmed by the results (e.g., how does the data science team evaluate these cases and update analysis and models to prevent future harm)?
 - [ ] **E.3 Roll back**: Is there a way to turn off or roll back the model in production if necessary?
 - [X] **E.4 Unintended use**: Have we taken steps to identify and prevent unintended uses and abuse of the model and do we have a plan to monitor these once the model is deployed?


## Team Expectations 

Team Expectation 1: Our team is committed to maintaining consistent communication, accountability, and respectful collaboration throughout the duration of the project. We will primarily communicate through Discord and text messaging, and all members agree to check messages at least twice daily to stay updated on deadlines, task assignments, and project progress. We will communicate any scheduling conflicts, delays, ask questions or troubleshoot any issues. We will check-in at 8 P.M. the day before, and on the date of submission, for a particular group assignment. 

Team Expectation 2: To ensure equal participation, responsibilities will be divided based on workload, individual strengths, and availability, while still ensuring that all members contribute to the major components of the project, including research, coding, data analysis, writing, and presentation preparation. Drafts and assigned tasks will be completed by internally agreed-upon deadlines to allow sufficient time for peer review and revisions before final submission. All code and written work will be shared through GitHub to maintain organization, version control, and transparency of contributions.

Team Expectation 3: Our team will conduct daily asynchronous check-ins where members provide updates on completed work, ongoing challenges, and goals for the day. In addition, we will hold regular meetings to review progress, troubleshoot issues, and coordinate upcoming tasks. We agree to provide constructive feedback respectfully and maintain a supportive working environment where all members feel comfortable asking questions or requesting help.

Team Expectation 4: If disagreements arise, we will first attempt to resolve them through open discussion and compromise with all team members. If repeated missed responsibilities or lack of participation arise, the team will privately address the issue with the member first before escalating concerns to course staff if necessary. Our goal is to create an organized and collaborative team dynamic that allows all members to contribute meaningfully.  


## Project Timeline Proposal

| Meeting Date | Meeting Time | Completed Before Meeting | Discuss at Meeting |
|---|---|---|---|
| 5/12 | 8 PM | Data Checkpoint 1 completed | Review and discuss any changes to the Data Checkpoint 1 document |
| 5/13 | 8 PM | Complete full Data Checkpoint 1 by pushing changes into GitHub | Update the team on any possible changes or issues as they arise |
| 5/19 | 8 PM | Explore possible EDA for the appropriate dataset | Plan and discuss possible EDA to be used |
| 5/26 | 8 PM | Complete EDA and start analysis | Review results and connect them to the hypothesis |
| 6/2 | 8 PM | Complete EDA/modeling | Edit the full project and improve visuals |
| 6/9 | 8 PM | Finalize report and prepare video | Record videos and submit final report |